src/game/GameManager.cpp

```diff
@@ -258,30 +258,34 @@ void GameManager::deleteObject(const int32_t id) {
     objectManager.erase(id);
 }
 
+//Created By Maitiu
+//Adds Weapon to Weapon Manager
 int32_t GameManager::addWeapon(std::shared_ptr<Weapon> weapon){
-    const int32_t id = weapon->getId();
+    const int32_t id = weapon->getID();
     const auto& elem = weaponManager.emplace(id, weapon);
     elem->second->setId(id);
     return id;
 }
 
+//Created By Maitiu
 int32_t GameManager::addWeaponDrop(WeaponDrop& newWeaponDrop) {
     const int32_t id = newWeaponDrop.getId();
     weaponDropManager.emplace(id, newWeaponDrop);
     return id;
 }
 
 // Create weapon drop add it to manager, returns success
-bool GameManager::createWeaponDrop(const float x, const float y) {
-    Rifle w;
-    const int32_t wid = w.getId();
+//Created By Maitiu
+bool GameManager::createWeaponDrop(const float x, const float y, int32_t wID) {
     const int32_t id = generateID();
 
     SDL_Rect weaponDropRect = {static_cast<int>(x),static_cast<int>(y), DEFAULT_SIZE, DEFAULT_SIZE};
     SDL_Rect pickRect = {static_cast<int>(x),static_cast<int>(y), DEFAULT_SIZE, DEFAULT_SIZE};
 
-    addWeapon(std::dynamic_pointer_cast<Weapon>(std::make_shared<Rifle>(w)));
-    weaponDropManager.emplace(id, WeaponDrop(id, weaponDropRect, pickRect, wid));
+    weaponDropManager.emplace(id, WeaponDrop(id, weaponDropRect, pickRect, wID));
+    getWeaponDrop(id).setPosition(x, y);
```

Comment:
Use return value of emplace for the set position call instead of calling a getter
One less function call